truss/remote/baseten/core.py

```diff
@@ -118,15 +147,22 @@ def create_chain(
                 raise ValueError(
                     f'Environment "{environment}" does not exist. You can create environments in the Chains UI.'
                 ) from e
+
             raise e
+    elif environment and environment != PRODUCTION_ENVIRONMENT_NAME:
+        raise ValueError(NO_ENVIRONMENTS_EXIST_ERROR_MESSAGING)
     else:
-        if environment and environment != PRODUCTION_ENVIRONMENT_NAME:
-            raise ValueError(NO_ENVIRONMENTS_EXIST_ERROR_MESSAGING)
-        response = api.deploy_chain(chain_name, chainlets)
+        res = api.deploy_chain_atomic(
+            chain_name=chain_name,
+            entrypoint=entrypoint,
+            dependencies=dependencies,
+        )
```

Comment:
How about the following restructure for better readability and less repetitions of `eploy_chain_atomic` invocations:

* Prepare all arguments to `eploy_chain_atomic`.
* Validate argument values/combinations.
* Make a single unified call to  `deploy_chain_atomic` (maybe modify the exception handling for missing env - is there are more specific message in the reponse to filter for?)

Reply:
That sounds good. I don't feel super safe making that change at this point but I'll add a `TODO`.